spec/requests/api/v0/university_registrations_spec.rb

```diff
@@ -20,12 +20,24 @@
     end
 
     it 'JSON body response contains a role' do
-      expect(response_json['data']['role']).to eq 'research_group'
+      expect(response_json['data']['user']['role']).to eq 'university'
     end
+  end
 
-    it 'JSON body response contains a name ' do
-      expect(response_json['data']['name']).to eq 'Fat Bob'
+  describe 'with valid credentials for Research Group role' do
+    before 'post new Research Group info' do
+      post '/api/v0/auth', params: { email: 'example@craftacademy.se',
+                                    name: 'Fat Bob',
+                                    role: 'research_group',
+                                    password: 'password',
+                                    password_confirmation: 'password' },
+                                    headers: headers
     end
+
+    it 'returns zero registration keys' do
+      expect(response_json['data']['registration_keys'].count).to eq 0
```

Comment:
Is the purpose of the entire `describe` to make sure that we don't generate registration keys? In that case, you probably want to make sure that you actually use the new way of registering a `reserch_group` - meaning that you have to use a key to connect it to a university. Does that make sense? This spec actually exposes a flaw in your implementation allowing a `reserch_group` to register without a key.

Reply:
Removed !